graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/tweet/TopSecondDegreeByCountForTweet.java

```diff
@@ -55,9 +56,51 @@ public TopSecondDegreeByCountForTweet(
     super(leftIndexedBipartiteGraph, expectedNodesToHit, statsReceiver);
   }
 
+  protected long getEdgeTimeStampInMillis(
+    long timeStampFromTweetId,
+    byte edgeType,
+    long edgeMetadata
+  ) {
+    switch (edgeType) {
+      case 0:  // CLICK
+        return edgeMetadata;
+      case 1:  // FAVORITE
+        return edgeMetadata;
+      case 2:  // RETWEET
+        return RecentTweetFilter.timeStampFromTweetId(edgeMetadata);
```

Comment:
1. For favorites, the edge timestamp is the timestamp of the Favorite.
Yes. 
2. For retweets, replies etc, the edge timestamp is the timestamp of the original tweet
For retweets, replies and quotes, the edge metadata is the intermediate tweet id, which is a SnowFlake id, and we can decode the timestamp from the SnowFlake id.

3. Why do we treat isMentioned and isMediatagged separately? Can't we just use RecentTweetFilter.timeStampFromTweetId for consistency?
For mention, mediaTag and create, right now, both RHS node and metadata store tweet ids. However, we might change the content of metadata later, but RHS node must always be tweet id. Here I use RHS node to decode timestamp instead of metadata to be future proof.